### PR TITLE
First Review mmMultiCR::UpdateGame()

### PR DESCRIPTION
**Pull request by KcRobin9**

### CODE REVIEW

File: code/midtown/mmaudio/sound.h [hunk 1]
```diff
@@ -91,6 +91,8 @@
 
 #include "arts7/node.h"
 
+#include "mmai/aiGoalFollowWayPts.h"
```
Comment (0x1F9F1) [inline code/midtown/mmaudio/sound.h:94]: Why is this included here?

File: code/midtown/mmgame/wpobject.h [hunk 2]
```diff
@@ -37,6 +37,8 @@
 
 #include "arts7/node.h"
 
+#include "mmai/aiGoalFollowWayPts.h"
```
Comment (0x1F9F1) [inline code/midtown/mmgame/wpobject.h:40]: Why is this included here?

File: code/midtown/pcwindis/dxinit.cpp [hunk 3]
```diff
@@ -220,7 +220,7 @@ Ptr<agiSurfaceDesc> dxiScreenShot()
     Ptr<agiSurfaceDesc> surface =
         as_ptr agiSurfaceDesc::Init(width, height, agiSurfaceDesc::FromFormat(PixelFormat_B8G8R8));
 
-    void (*translate)(u8 * output, u16 * input, u32 width) = nullptr;
+    void (*translate)(u8* output, u16* input, u32 width) = nullptr;
```
Comment (0x1F9F1) [inline code/midtown/pcwindis/dxinit.cpp:223]: Unfortunately different versions of clang-format can produce slightly different results. Since this isn't part of the code you changed can you revert this?

File: code/midtown/mmaudio/sound.h [hunk 4]
```diff
@@ -275,7 +277,7 @@ class AudSound final : public asNode
     // ?GetSoftFreqChange2DFlags@AudSound@@SA?BKXZ | unused
     ARTS_IMPORT static const ulong GetSoftFreqChange2DFlags();
 
-private:
+public:
```
Comment (0x1F9F1) [inline code/midtown/mmaudio/sound.h:280]: I'd rather leave this as private, then add a `public` specifically for the member variables

File: code/midtown/mmaudio/sound.h [hunk 5]
```diff
@@ -296,7 +298,40 @@ class AudSound final : public asNode
 
     ARTS_DIAGNOSTIC_POP;
 
-    u8 gap20[0x98];
+    i32 field_20;
+    i32 Priority;
+    i32 dword28;
```
Comment (0x1F9F1) [inline code/midtown/mmaudio/sound.h:303]: Can you update the unknown members to be `field_<offset>`?

File: code/midtown/mmgame/multicr.cpp [hunk 6]
```diff
@@ -51,3 +61,136 @@ i32 mmMultiCR::SelectTeams()
 
 void mmMultiCR::SendSetup(ulong /*arg1*/)
 {}
+
+void mmMultiCR::StealGold(mmCar* car)
+{
+    GoldCarrier = car;
+
+    mmMultiCR::FondleCarMass(GoldCarrier, float(MMSTATE.CRGoldMass));
```
Comment (0x1F9F1) [inline code/midtown/mmgame/multicr.cpp:69]: Use `static_cast<float>` instead

File: code/midtown/mmgame/multicr.cpp [hunk 7]
```diff
@@ -51,3 +61,136 @@ i32 mmMultiCR::SelectTeams()
...
+    field_1EEB0->Initialized = true;
+}
+
+void mmMultiCR::UpdateGame()
+{
+    switch (GameState)
```
Comment (0x1F9F1) [inline code/midtown/mmgame/multicr.cpp:75]: Add braces around each of the switch cases

File: code/midtown/mmgame/multicr.cpp [hunk 8]
```diff
@@ -51,3 +61,136 @@ i32 mmMultiCR::SelectTeams()
...
+                if (VoiceCommentary)
+                    VoiceCommentary->PlayTimePenalty();
+
+                GameStateWait = 5.0f;
+                GameState = 6;
+
+                if (GoldCarrier == &Player->Car)
+                {
+                    DropGold(Player->Car.GetICS()->Matrix.m3, 0);
+                    FondleCarMass(&Player->Car, -float(MMSTATE.CRGoldMass));
```
Comment (0x1F9F1) [inline code/midtown/mmgame/multicr.cpp:122]: Use `static_cast<float>`

File: code/midtown/mmgame/multicr.cpp [hunk 9]
```diff
@@ -51,3 +61,136 @@ i32 mmMultiCR::SelectTeams()
...
+            if (MMSTATE.CRLimitMode == mmCRLimitMode::Time && NETMGR.IsHost())
+            {
+                field_1EEFC.Start();
+                Player->Hud.BlitzTimer.Start();
+            }
+
+            StartSounds->ActiveSound = 0;
+            StartSounds->PlayOnce(-1.0, -1.0);
+
+            Player->Hud.SetMessage(AngelReadString(0x83u), 2.0f, true);
```
Comment (0x1F9F1) [inline code/midtown/mmgame/multicr.cpp:99]: Use `LOC_STR(MM_IDS_183)` (but also rename MM_IDS_183 to something more descriptive)

File: code/midtown/mmgame/multicr.cpp [hunk 10]
```diff
@@ -51,3 +61,136 @@ i32 mmMultiCR::SelectTeams()
...
+                VoiceCommentary->PlayCRPreRace();
+
+            EnableRacers();
+            GameState = 4;
+            break;
+
+        case 4:
+            if (!MMSTATE.DisableDamage && Player->IsMaxDamaged())
+            {
+                Player->Hud.SetMessage(AngelReadString(0x84u), 5.0f, false);
```
Comment (0x1F9F1) [inline code/midtown/mmgame/multicr.cpp:111]: Use `LOC_STR(MM_IDS_184)` (but also rename MM_IDS_184 to something more descriptive)

File: powershell.exe
Comment (0x1F9F1) [inline powershell.exe:1]: Remove this.

File: code/midtown/mmaudio/sound.h (hunk 1 above)
Comment (KcRobin9) [reply code/midtown/mmaudio/sound.h:94]: If omitted we will get "incomplete type is not allowed" for:
    Vector3 LastPosition;
    Vector3 Velocity;

File: code/midtown/mmgame/wpobject.h (hunk 2 above)
Comment (KcRobin9) [reply code/midtown/mmgame/wpobject.h:40]: I will remove this one

File: code/midtown/mmaudio/sound.h (hunk 1 above)
Comment (0x1F9F1) [reply code/midtown/mmaudio/sound.h:94]: Then you need to include the header for Vector3 (`vector7/vector3.h`)

File: code/midtown/mmgame/multicr.cpp [hunk 11]
```diff
@@ -51,3 +61,145 @@ i32 mmMultiCR::SelectTeams()
...
+            EnableRacers();
+            GameState = 4;
+            break;
+        }
+
+        case 4: {
+            if (!MMSTATE.DisableDamage && Player->IsMaxDamaged())
+            {
+                Player->Hud.SetMessage(LOC_STRING(MM_IDS_DAMAGE_PENALTY), 5.0f, false);
+                ;
```
Comment (0x1F9F1) [inline code/midtown/mmgame/multicr.cpp:115]: Extra semi-colon

File: code/midtown/mmgame/multicr.cpp [hunk 12]
```diff
@@ -51,3 +61,145 @@ i32 mmMultiCR::SelectTeams()
...
+            }
+            break;
+        }
+
+        case 10: {
+            if (!Popup->IsEnabled())
+                Popup->ShowResults();
+            break;
+            default: break;
+        }
```
Comment (0x1F9F1) [inline code/midtown/mmgame/multicr.cpp:187]: The default case should outside the braces
